_studio/shared/umc/codec/vc1_dec/include/umc_vc1_video_decoder_hw.h

```diff
@@ -157,10 +157,22 @@ namespace UMC
 
             m_bLastFrameNeedDisplay = true;
 
-            if (VC1_SKIPPED_FRAME == pPackDescriptorChild->m_pContext->m_picLayerHeader->PTYPE)
-                out_data->SetFrameType(D_PICTURE); // means skipped 
-            else
-                out_data->SetFrameType(I_PICTURE);
+            switch (pPackDescriptorChild->m_pContext->m_picLayerHeader->PTYPE)
+            {
+                case  VC1_I_FRAME:
+                    out_data->SetFrameType(I_PICTURE);
+                    break;
+                case  VC1_P_FRAME:
+                    out_data->SetFrameType(P_PICTURE);
+                    break;
+                case  VC1_B_FRAME:
+                case  VC1_BI_FRAME:
+                    out_data->SetFrameType(B_PICTURE);
+                    break;
+                case  VC1_SKIPPED_FRAME:
+                    out_data->SetFrameType(D_PICTURE); // means skipped
+                    break;
```

Comment:
default is missing. Can we potentially come into it here?

Reply:
Default value (NONE) sets on initialization  out_data, we can reset NONE here, but I guess it's extra action